### ISSUE TITLE
Cannot find name 'Deno'. deno-ts(2304)

### ISSUE THREAD

**Issue by JHarrisGTI** [denoland/vscode_deno#841]
I'm opening a file in a previously-working project. When I call `Deno.readFile`, "Deno" gets a red squiggly underline with this message:

`Cannot find name 'Deno'. deno-ts(2304)`

My .vscode/settings.json is:

```json
{
    "deno.enable": true,
    "deno.unstable": true
}
```

I've read #66 and #466 looking for fixes. I'm using Deno 1.32.5, TypeScript 5.0.3, and VSCode extension 3.17.0 on MacOS 13.3.1 (Intel). Things I've tried:
- Uninstalling and reinstalling the VSCode plugin
- Restarting the language server
- Restarting VSCode
- Deleting my .vscode/settings.json and running `Deno: Initialize Workspace Configuration`
- Disabling/reenabling `deno.enable`
- Disabling/reenabling `deno.unstable`
- Disabling/reenabling `deno.lint` 
- Running `brew upgrade deno`

I don't have a tsconfig.json file or anything else; the TypeScript file I'm working on is standalone aside from importing an NPM library and reading/writing some PDF files.

<details>
<summary>Deno language server output looks fine.</summary>
<pre>
Starting Deno language server...
  version: 1.32.5 (release, x86_64-apple-darwin)
  executable: /usr/local/bin/deno
Connected to "Visual Studio Code" 1.77.3
Enabling import suggestions for: https://crux.land
Enabling import suggestions for: https://x.nest.land
Download https://x.nest.land/.well-known/deno-import-intellisense.json
Download https://intellisense.nest.land/deno-import-intellisense.json
Enabling import suggestions for: https://deno.land
Server ready.
</pre>
</details>

**Comment by alexandruradovici** [comment]
+1

**Comment by captainbuckkets** [comment]
+1

**Comment by captainbuckkets** [comment]
https://github.com/denoland/vscode_deno/issues/66

**Comment by denyzhirkov** [comment]
+

**Comment by bartlomieju** [comment]
We are investigating this issue. 

Current hypothesis is that the problem manifests itself when "triple-slash directives" are used:
```js
/// <reference lib="foo" />
```

A temporary workaround should be to add:
```
/// <reference lib="deno.ns" />
```

That said we are investigating a better solution that at least suggests users to add that directive in certain scenarios.

**Comment by JHarrisGTI** [comment]
After some experimentation, I've determined that in my project, the problem is caused by this line of code:

```ts
import { EmbedPdf } from "https://deno.land/x/embed_pdf@v1.3.0/mod.js";
```

Commenting out this line and using VSCode's "Developer: Reload Window" command suppresses the problem.

[That file](https://github.com/ayame113/embed-pdf-element/blob/main/mod.js) does indeed have triple-slash directives in it, so I think you're on the right track.

**Comment by bartlomieju** [comment]
FYI https://github.com/denoland/deno/pull/19825 improves the diagnostics and was released in Deno v1.35.2. Please try it and let me know if that helps.

**Comment by JHarrisGTI** [comment]
I'm now seeing this error message:

> Cannot find name 'Deno'. Do you need to change your target library? Try changing the 'lib' compiler option to include 'deno.ns' or add a triple-slash directive to your entrypoint: `/// <reference lib="deno.ns" />`

The more detailed error message is helpful, thank you! I'm not sure where I would change my compiler options, or what "entrypoint" means in this context. 

Adding that directive to the file I'm working on does not reliably fix the problem, I'm afraid.

For reference, here's the file I'm working on.

```ts

import { readCSVObjects } from "https://deno.land/x/csv@v0.8.0/mod.ts";
import { EmbedPdf } from "https://deno.land/x/embed_pdf@v1.3.0/mod.js";

/// <reference lib="deno.ns" />

export async function getPayload() {
const file = await Deno.open("./test payload.csv");

	let payload;
	for await (const obj of readCSVObjects(file, { lineSeparator: "\r" })) {
		payload = obj;
	}

	file.close();

	return payload;
}
```

**Comment by dsherret** [comment]
@JHarrisGTI move the `/// <reference lib="deno.ns" />` to the top of the file. I'll update the message.

The issue is happening because of that library is doing this:

https://github.com/ayame113/embed-pdf-element/blob/main/mod.js#L3

**Comment by JHarrisGTI** [comment]
Yes, moving the directive to the top of the file fixes the issue. And the new message makes that clear. Thank you for fixing this issue! 🍺
